llvm/lib/Support/APFloat.cpp

```diff
@@ -4626,6 +4742,11 @@ void IEEEFloat::makeInf(bool Negative) {
 }
 
 void IEEEFloat::makeZero(bool Negative) {
+  // The E8M0 type cannot represent the value zero.
+  if (semantics == &semFloat8E8M0FN) {
+    assert(false && "This floating point format does not support Zero\n");
```

Comment:
Firstly, assert(false) shouldn't be in the code base. To me it sounds more like a no-op.

Reply:
Agreed, Converted it to an llvm_unreachable().